Library/Homebrew/formula.rb

```diff
@@ -1486,20 +1487,40 @@ def recursive_requirements(&block)
 
   # Returns a list of Dependency objects that are required at runtime.
   # @private
-  def runtime_dependencies(read_from_tab: true)
-    if read_from_tab &&
-       installed_prefix.directory? &&
-       (keg = Keg.new(installed_prefix)) &&
-       (tab_deps = keg.runtime_dependencies)
+  def runtime_dependencies
+    if installed_prefix.directory? &&
+       (tab_deps = Keg.new(installed_prefix)&.runtime_dependencies)
+
       return tab_deps.map { |d| Dependency.new d["full_name"] }.compact
     end
 
+    compute_runtime_dependencies
+  end
+
+  def compute_runtime_dependencies
```

Comment:
Sorry to be a pain but I think this is probably worse than the `read_from_tab` argument was 😭. It's not clear when you need to call `runtime_dependencies` vs. `compute_runtime_dependencies`. Could maybe fiddle with naming options but I'm tempted to just say put it back to how it was. Sorry.